piff/simplepsf.py

```diff
@@ -178,6 +178,10 @@ def fit(self, stars, wcs, pointing,
             if hasattr(self.model, 'reflux'):
                 new_stars = []
                 for s in self.stars:
+                    # Update the noise with the new interpolated signal.
+                    signal = self.drawStar(s)
+                    s = s.addPoisson(signal)
```

Comment:
We don't add noise to the image, just the weight map.  It takes the original variance from the input weight map and adds the (signal image / gain) to the variance.  It's better to do this with the model signal rather than the observed data (as we had been doing), since the latter can lead to biases.